vSphere Tutorials Take #2

cormachogan opened the pull request:
Adding vSphere Tutorials

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-Stateful-Backup-vSphere.md:
````diff
@@ -0,0 +1,460 @@
+cqlsh:demodb>
+```
+
+It looks like the restore has been successful. Velero v1.1 has successfully restored the Kubenetes objects for the Cassandra application, as well as restored the database and table contents.
````
```suggestion
It looks like the restore has been successful. Velero v1.1 has successfully restored the Kubenetes objects for the Cassandra application, as well as restored the database and table of contents.
```

cormachogan replied:
No, it is not a table of contents, it is the contents of the table. This is ok as is.

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-Stateful-Backup-vSphere.md:
```diff
@@ -0,0 +1,460 @@
+
+## Join the Movement – Contribute
+
+Velero is better because of our contributors and maintainers. It is because of them that we can bring great software to the community. Please join us during our [online community meetings every first Tuesday](https://github.com/heptio/velero-community) and catch up with past meetings on YouTube on the [Velero Community Meetings playlist](https://www.youtube.com/watch?v=nc48ocI-6go&list=PL7bmigfV0 ...
```
It's awesome you included this! I'd like to avoid being specific going forwards because some of these things have been changing.

Best is to point to our community page, which is the entry point for anyone new: 
https://deploy-preview-1900--velero.netlify.com/community/

Feel free to copy the wording from this PR: https://github.com/heptio/velero/pull/1897/files#diff-c933f84352acf96ca21a8c9249d5bfa1R49

And same for the other 2 posts please.

cormachogan replied:
Done. I did leave in both the slack channel and twitter handle. Always useful imo.

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,352 @@
+
+## Deploy and Configure a Minio Object Store as a backup destination
+
+Velero sends data and metadata about the Kubernetes objects being backed up to an S3 Object Store. If you do not have an S3 Object Store available, Velero provides the manifest file to create a Minio S3 Object Store on your Kubernetes cluster. This means that all Velero backsup can be kept on-premises.
```
```suggestion
Velero sends data and metadata about the Kubernetes objects being backed up to an S3 Object Store. If you do not have an S3 Object Store available, Velero provides the manifest file to create a Minio S3 Object Store on your Kubernetes cluster. This means that all Velero backups can be kept on-premises.
```

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,352 @@
+
+While this step is optional, it is useful for two reasons. The first is that it gives you a way to access the Minio portal through a browser and examine the backups. The second is that it enables you to specify a `publicUrl` for Minio, which in turn means that you can access backup and restore logs from the Minio S3 Object Store.
+
+To expose the Minio Service on a NodePort, a modification of the `examples/minio/00-minio-deployment.yaml` manifest is necessary. The only change is to the type: field, from ClusterIP to NodePort:
```
I think this needs to be done *after* it is applied, by editing the deployment. I forget why or even if this is correct. @skriss would you please confirm one way or the other?

skriss replied:
Nope, you can change this in the YAML before applying it to the cluster.

cormachogan replied:
Cool - thx @skriss - yeah, there are the instructions I followed and they worked just fine.

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md:
```diff
@@ -0,0 +1,261 @@
+
+While this step is optional, it is useful for two reasons. The first is that it gives you a way to access the Minio portal through a browser and examine the backups. The second is that it enables you to specify a `publicUrl` for Minio, which in turn means that you can access backup and restore logs from the Minio S3 Object Store.
+
+To expose the Minio Service on a NodePort, a modification of the `examples/minio/00-minio-deployment.yaml` manifest is necessary. The only change is to the type: field, from ClusterIP to NodePort:
```
See my comment on the other post about editing the deployment.

cormachogan replied:
Yep - @skriss  confirmed this is how to do it. Also worked this way in my testing.

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md:
```diff
@@ -0,0 +1,261 @@
+
+## Deploy a sample application to backup
+
+Velero provide a sample `nginx` application for backup testing. This nginx deployment assumes the presence of a LoadBalancer for its Service. If you do not have a Load Balancer as part of your Container Network Interface (CNI), there are some easily configuration ones available to get your started. One example is MetalLb, available [here](https://metallb.universe.tf/).
```
```suggestion
Velero provides a sample `nginx` application for backup testing. This nginx deployment assumes the presence of a LoadBalancer for its Service. If you do not have a Load Balancer as part of your Container Network Interface (CNI), there are some easily configuration ones available to get your started. One example is MetalLb, available [here](https://metallb.universe.tf/).
```

cormachogan replied:
Done

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md:
```diff
@@ -0,0 +1,261 @@
+
+## Deploy and Configure a Minio Object Store as a backup destination
+
+Velero sends data and metadata about the Kubernetes objects being backed up to an S3 Object Store. If you do not have an S3 Object Store available, Velero provides the manifest file to create a Minio S3 Object Store on your Kubernetes cluster. This means that all Velero backsup can be kept on-premises.
```
```suggestion
Velero sends data and metadata about the Kubernetes objects being backed up to an S3 Object Store. If you do not have an S3 Object Store available, Velero provides the manifest file to create a Minio S3 Object Store on your Kubernetes cluster. This means that all Velero backups can be kept on-premises.
```

cormachogan replied:
Fixed.

carlisia commented on site/_posts/2019-09-20-Velero-v1-1-Stateful-Backup-vSphere.md:
```diff
@@ -0,0 +1,460 @@
+---
+title: Velero v1.1 backing up and restoring Stateful apps on vSphere and PKS (Pivotal Container Service)
+excerpt: This post demonstrates how Velero can be used on PKS, the Pivotal Container Service which deploys Kubernetes clusters on vSphere.
```
Any chance you could have an image to use for this post, so we don't default to the Velero logo?

cormachogan replied:
Added Cassandra logo as that is the app I am backing up and restoring.

cormachogan commented on site/_posts/2019-09-20-Velero-v1-1-Stateful-Backup-vSphere.md:
```diff
@@ -0,0 +1,460 @@
+title: Velero v1.1 backing up and restoring Stateful apps on vSphere and PKS (Pivotal Container Service)
+excerpt: This post demonstrates how Velero can be used on PKS, the Pivotal Container Service which deploys Kubernetes clusters on vSphere.
+author_name: Cormac Hogan
+author_avatar: ../img/contributors/cormac-pic.png
```
Do you mean that I should just remove the .. from the ../img path?

cormachogan replied:
You did  :-) Looks like it is working after the changed.

cormachogan commented on site/_posts/2019-09-20-Velero-v1-1-Stateful-Backup-vSphere.md:
```diff
@@ -0,0 +1,460 @@
+
+## Destroy the Cassandra deployment
+
+Now that we have successfully take a backup, which includes the `Restic` backups of the data, we will now go ahead and destroy the Cassandra namespace, and restore it once again.
```
Fixed

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,350 @@
+# Tag should match author to drive author pages
+tags: ['Velero', 'Cormac Hogan', 'how-to']
+---
+Velero version 1.1 provides support to backup Kubernetes applications orchestrated on top of PKS, the Pivotal Container Service for vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in a Kubernetes cluster deployed by PKS on vSphere. At this time there is no vSphere plugin for snapsho ...
```
The `here` link at the end doesn't seem to be working in the site preview - we probably want to make it an absolute  URL, e.g. `velero.io/velero-v1-1-stateful-backup-vsphere/`

cormachogan replied:
Fixed

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,350 @@
+
+In the above outputs, the node on which the Minio Object Storage is deployed has IP address `192.168.192.5`. The NodePort that the Minio Service is exposed is `32109`. If we now direct a browser to that `Node:port` combination, we should see the Minio Object Store web interface. You can use the credentials provided in the `credentials-velero` file earlier to login.
+
+![Minio Object Store](../img/vsphere-tutorial-icons/Minio.png)
```
this image ref is broken in the site preview - looks like the file doesn't exist.

cormachogan replied:
Good catch - forgot to include the folder when we re-created the PR.

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,350 @@
+
+In this example, a Load Balancer has provided the `nginx` service with an external IP address of 192.168.191.70. If I point a browser to that IP address, I get an nginx landing page identical to that shown below.
+
+![nginx landing page](../img/vsphere-tutorial-icons/nginx.png)
```
this image link is also broken - I don't see this file in the PR

cormachogan replied:
Good catch - forgot to include the folder when we re-created the PR.
Fixed now (also removed the ../ and referenced /img directly)

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,350 @@
+
+You can now login to the Minio Object Storage via a browser and verify that the backup actually exists. You should see the name of the backup under the `velero/backups` folder:
+
+![Minio Backup Details](../img/vsphere-tutorial-icons/minio-nginx-backup.png)
```
missing file

cormachogan replied:
Fixed

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,350 @@
+
+Now let’s see if we can successfully reach our `nginx` web server on that IP address. Yes we can! Looks like the restore was successful.
+
+![nginx restored](../img/vsphere-tutorial-icons/nginx-restore-new-ip.png)
```
missing file

cormachogan replied:
Fixed

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md:
```diff
@@ -0,0 +1,259 @@
+# Tag should match author to drive author pages
+tags: ['Velero', 'Cormac Hogan', 'how-to']
+---
+Velero version 1.1 provides support to backup Kubernetes applications deployed on vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in Kubernetes on vSphere. At this time there is no vSphere plugin for snapshotting stateful applications on vSphere during a Velero backup. In this case, ...
```
Similar to previous post, the `here` link is broken and we probably want the `velero.io/...` link I put in the prior comment.

cormachogan replied:
Fixed

skriss commented on site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md:
```diff
@@ -0,0 +1,259 @@
+
+In the above outputs, the node on which the Minio Object Storage is deployed has IP address `192.168.192.5`. The NodePort that the Minio Service is exposed is `32109`. If we now direct a browser to that `Node:port` combination, we should see the Minio Object Store web interface. You can use the credentials provided in the `credentials-velero` file earlier to login.
+
+![Minio Object Store](../img/vsphere-tutorial-icons/Minio.png)
```
the image links are all broken on this page too - missing files.

cormachogan replied:
Fixed

skriss commented:
BTW, here's the "site preview" link for this PR: https://deploy-preview-1900--velero.netlify.com

jonasrosland commented:
> Do we want all these posts to go up at once, or should we stagger them? @jonasrosland any input here?

Definitely stagger them. We've got three blog posts here, tackling similar but different subject. Great content! Thank you @cormachogan. Please /hold as I'll review them as well.

skriss commented:
> Definitely stagger them. We've got three blog posts here, tackling similar but different subjects. Great content! Thank you @cormachogan. Please /hold as I'll review them as well.

@jonasrosland can we accomplish this by just setting the dates on some of the posts to the future, and then merging them all together?

jonasrosland commented:
> > Definitely stagger them. We've got three blog posts here, tackling similar but different subjects. Great content! Thank you @cormachogan. Please /hold as I'll review them as well.
> 
> @jonasrosland can we accomplish this by just setting the dates on some of the posts to the future, and then merging them all together?

Yup, that should do it.

cormachogan commented:
OK - Let me stagger the dates of the posts, starting from Friday (27) to give people a chance to review.

skriss commented:
I guess unfortunately now we can't see them in the site preview :/

cormachogan commented:
Ah, OK - let me change them back and make sure the missing files are now present. Then I can reset them to the future when everyone is happy with them.

skriss commented on site/_posts/2019-09-27-Velero-v1-1-on-vSphere.md:
```diff
@@ -8,7 +8,7 @@ categories: ['kubernetes']
 tags: ['Velero', 'Cormac Hogan', 'how-to']
 ---
-Velero version 1.1 provides support to backup Kubernetes applications deployed on vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in Kubernetes on vSphere. At this time there is no vSphere plugin for snapshotting stateful applications on vSphere during a Velero backup. In this case, ...
+Velero version 1.1 provides support to backup Kubernetes applications deployed on vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in Kubernetes on vSphere. At this time there is no vSphere plugin for snapshotting stateful applications on vSphere during a Velero backup. In this case, ...
```
Two things:
- if we publish this post first, the  `here` link will be broken until the post it points to gets published. We could reorder the post publish dates, or remove the link and just say "...a future post", or accept the broken link for a few days.
- the `.md` needs to be dropped from the end of the link URL

cormachogan replied:
Done

skriss commented on site/_posts/2019-09-30-Velero-v1-1-on-PKS.md:
```diff
@@ -8,7 +8,7 @@ categories: ['kubernetes']
 tags: ['Velero', 'Cormac Hogan', 'how-to']
 ---
-Velero version 1.1 provides support to backup Kubernetes applications orchestrated on top of PKS, the Pivotal Container Service for vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in a Kubernetes cluster deployed by PKS on vSphere. At this time there is no vSphere plugin for snapsho ...
+Velero version 1.1 provides support to backup Kubernetes applications orchestrated on top of PKS, the Pivotal Container Service for vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in a Kubernetes cluster deployed by PKS on vSphere. At this time there is no vSphere plugin for snapsho ...
```
Same comment re: publishing date and broken link.

cormachogan replied:
I could do the vSphere one and the Cassandra one at the same time and avoid the missing link? Then publish the PKS one a few days later. Will remove the .md now.

skriss replied:
That works for me!

cormachogan replied:
Coolio. OK, not sure who else needs to review, but as soon as, let me know and I will change the dates of the filenames and do one final push. Thanks for all the feedback.

cormachogan commented:
Once reviews are complete, please reach out and I will change the dates in the filenames for one final push before merge. Thanks.

skriss commented on site/_posts/2019-09-24-Velero-v1-1-on-PKS.md:
```diff
@@ -0,0 +1,350 @@
+# Tag should match author to drive author pages
+tags: ['Velero', 'Cormac Hogan', 'how-to']
+---
+Velero version 1.1 provides support to backup Kubernetes applications orchestrated on top of PKS, the Pivotal Container Service for vSphere. This post will provide detailed information on how to install and configure Velero to backup and restore a stateless application (`nginx`) that is running in a Kubernetes cluster deployed by PKS on vSphere. At this time there is no vSphere plugin for snapsho ...
```
argh, looks like the `here` link is still busted. The following format seems to work for me testing locally:

```
[here](../Velero-v1-1-Stateful-Backup-vSphere/)
```

cormachogan replied:
Put them back the way they were - should hopefully work now.

jonasrosland commented:
Hey @cormachogan I see that the PKS and vSphere blog posts are almost identical, with minor changes between them. Could you merge them into one called "Velero v1.1 backing up and restoring apps on Kubernetes on vSphere" or similar?

cormachogan commented:
> Hey @cormachogan I see that the PKS and vSphere blog posts are almost identical, with minor changes between them. Could you merge them into one called "Velero v1.1 backing up and restoring apps on Kubernetes on vSphere" or similar?

I did consider doing that @jonasrosland but I think it makes sense to keep them separate. I think trying to merge them, and then call out what is needed when it is PKS versus native K8s gets confusing in my opinion.

jonasrosland commented:
Is the only change the `hostPath`, is that correct?
If so then I think it's perfectly ok calling that out as a "special use case", but it doesn't really warrant its own full blog post.

skriss commented:
@jonasrosland @cormachogan did we resolve the publishing schedule/location for this?

cormachogan commented:
> @jonasrosland @cormachogan did we resolve the publishing schedule/location for this?

Waiting on @jonasrosland for a date - I believe the Tanzu stuff has taken precedence this week.

jonasrosland commented:
> > @jonasrosland @cormachogan did we resolve the publishing schedule/location for this?
> 
> Waiting on @jonasrosland for a date - I believe the Tanzu stuff has taken precedence this week.

Thank you for your patience here @cormachogan, really appreciate it.
To get these great articles posted, can you please do the following?

- [ ] Change the date on the vSphere blog to October 8
- [ ] Change the date on the Cassandra blog to October 10, and remove the PKS language as it looks like the instructions are the same as for a vanilla K8s environment? @skriss can you verify if that's correct or not?
- [ ] Remove the PKS blog from the PR as that will be posted elsewhere

cormachogan commented:
- Renamed setup and stateful tutorials to publish on 8th and 10th respectively.
- Removed PKS tutorial (not heard anything from CNA team btw)
- Removed PKS specific language from posts
- Modified some text to make one or two steps a bit clearer

jonasrosland commented on site/_posts/2019-10-08-Velero-v1-1-on-vSphere.md:
```diff
@@ -0,0 +1,259 @@
+
+## What this post does not show
+
+* How to install Velero v1.1 with PKS, Pivotal Container Service, on vSphere
```
Just a tiny nitpick, this line is unneeded.

cormachogan replied:
Done

jonasrosland commented on site/_posts/2019-10-10-Velero-v1-1-Stateful-Backup-vSphere.md:
```diff
@@ -0,0 +1,467 @@
+# Tag should match author to drive author pages
+tags: ['Velero', 'Cormac Hogan', 'how-to']
+---
+Velero version 1.1 provides support to backup applications orchestrated on upstream Kubernetes running natively on vSphere or on Kubernetes deployed by PKS, the Pivotal Container Service for vSphere. This post will provide detailed information on how to use Velero v1.1 to backup and restore a stateful application (`Cassandra`) that is running in a Kubernetes cluster deployed on vSphere. At this t ...
```
You can remove " or on Kubernetes deployed by PKS, the Pivotal Container Service for vSphere"

cormachogan replied:
Done

jonasrosland commented:
Looks great @cormachogan , thank you again for writing and sharing these!
